pkg/resourcesets/collector.go

```diff
@@ -255,6 +255,24 @@ func (h *ResourceHandler) filterByNameAndLabel(ctx context.Context, dr dynamic.R
 		}
 	}
 
+	// filter out using ExcludeResourceNameRegexp
+	if filter.ExcludeResourceNameRegexp != "" {
+		for _, resObj := range resourceObjectsList.Items {
+			metadata := resObj.Object["metadata"].(map[string]interface{})
+			name := metadata["name"].(string)
+			nameMatched, err := regexp.MatchString(filter.ExcludeResourceNameRegexp, name)
```

Comment:
If I'm understanding this correctly, this line should return `true` if the string in the variable `name` starts or ends with `default`, but I tried this in the go playground and got `false` instead. I do get that behavior if I use the pattern `^default|default$`, but I see we are using the `^<string>$` pattern to match other things already so am I missing something? https://go.dev/play/p/X1GyCu_9ymg

Reply:
The exclude is for service accounts called `default`, the playground shows comparing token names with the regex. We are excluding the default service accounts created in each namespace and they are named `default`. Let me know if I need to add some clarification in the issue or code.

Reply:
Okay, makes sense now. I think the comment you linked to is good, this is just my first time reviewing the backup-restore-operator.